lisa/util/shell.py

```diff
@@ -191,25 +200,30 @@ def mkdir(
         exist_ok: bool = False,
     ) -> None:
         path_str = self._purepath_to_str(path)
+        self.initialize()
         assert self._inner_shell
         self._inner_shell.mkdir(path_str, mode=mode, parents=parents, exist_ok=exist_ok)
 
     def exists(self, path: PurePath) -> bool:
+        self.initialize()
         assert self._inner_shell
         path_str = self._purepath_to_str(path)
         return cast(bool, self._inner_shell.exists(path_str))
 
     def remove(self, path: PurePath, recursive: bool = False) -> None:
+        self.initialize()
         assert self._inner_shell
         path_str = self._purepath_to_str(path)
         self._inner_shell.remove(path_str, recursive)
 
     def chmod(self, path: PurePath, mode: int) -> None:
+        self.initialize()
         assert self._inner_shell
         path_str = self._purepath_to_str(path)
         self._inner_shell.chmod(path_str, mode)
 
     def stat(self, path: PurePath) -> os.stat_result:
+        self.initialize()
         assert self._inner_shell
         path_str = self._purepath_to_str(path)
         sftp_attributes: paramiko.SFTPAttributes = self._inner_shell.stat(path_str)
```

Comment:
Unrelated to the patch, but SFTP can be disabled in SSH servers (or enabled only for certain users). Tests that rely on this will fail, which might be fine as we'll probably get a trace to address the configuration issue, but it might be a good idea to add a comment here so people investigating these issues in the future knows that this is the case.

Reply:
No sure what will happen here exactly. So let's leave it here, and comment behavior or improve when it happens. As we're testing new default systems, let's see if it happens on new default systems.

Reply:
Sure, sounds good